quickstart/src/main/java/com/twilio/video/quickstart/activity/VideoActivity.java

```diff
@@ -203,8 +198,17 @@ protected void onCreate(Bundle savedInstanceState) {
     public boolean onCreateOptionsMenu(Menu menu) {
         MenuInflater inflater = getMenuInflater();
         inflater.inflate(R.menu.menu_video_activity, menu);
-        turnSpeakerOnMenuItem = menu.findItem(R.id.menu_turn_speaker_on);
-        turnSpeakerOffMenuItem = menu.findItem(R.id.menu_turn_speaker_off);
+        audioDeviceMenuItem = menu.findItem(R.id.menu_audio_device);
+
+        /*
+         * Start the audio device selector after the menu is created and update the icon when the
+         * selected audio device changes.
+         */
+        audioDeviceSelector.start((audioDevices, audioDevice) -> {
+            updateAudioDeviceIcon(audioDevice);
+            return Unit.INSTANCE;
```

Comment:
I'd say this is the only API oddity from Kotlin to Java. It's not a big deal because the app will not compile if you do not return a Unit instance, but it took me a minute to figure out how to make the compiler happy. I wonder if we could add another `start` signature that takes a pure Java interface so that java developers do not have to return a unit type.

Reply:
Good feedback! Yes, we could add a Java interface to make it easier for java devs to consume. Here's another interesting solution by creating a wrapper in Java https://stackoverflow.com/a/37834971/5163090

Reply:
Good feedback! Yes, we could add a Java interface to make it easier for java devs to consume. Here's another interesting solution by creating a wrapper in Java https://stackoverflow.com/a/37834971/5163090